### ISSUE TITLE
Missing tests on XML report for JUnit suites on 3.0.8

### ISSUE THREAD

**Issue by gabrielrussoc** [scalatest/scalatest#1972]
## Problem
I have the following suite:
```
import org.junit.Assert;
import org.junit.Test;

public class ExampleSuite extends Assert {

    @Test
    public void success(){
    }

    @Test
    public void failure(){
	int x = 1/0;
    }
}
```

If I run it with `org.scalatest.tools.Runner.main(-j, ExampleSuite, -u, report, -oDF)`, following the [using the runner docs](https://www.scalatest.org/user_guide/using_the_runner), my tests run just fine:
```
Run starting. Expected test count is: 2
JUnitWrapperSuite:
- success
- failure *** FAILED ***
  java.lang.ArithmeticException: / by zero
  at ExampleSuite.failure(ExampleSuite.java:12)
 ... more stuff ...
Run completed in 111 milliseconds.
Total number of tests run: 2
Suites: completed 1, aborted 0
Tests: succeeded 1, failed 1, canceled 0, ignored 0, pending 0
*** 1 TEST FAILED ***
```

However, the generated `report/TEST-org.scalatest.junit.JUnitWrapperSuite.xml` has no `<testcase>` entries. The suite is also marked with 0 tests:
```
<testsuite
errors="0" failures="0" hostname="something" name="org.scalatest.junit.JUnitWrapperSuite" tests="0" time="0.032" timestamp="2021-02-25T11:59:58">
```

Full repro on https://github.com/gabrielrussoc/scalatest-junit-xml/tree/3.0.8

## Possible solution

On [Suite.scala](https://github.com/scalatest/scalatest/blob/3.0.x/scalatest/src/main/scala/org/scalatest/Suite.scala#L1283), we define `suiteId` to be the classname. For JUnit tests, we use a wrapper suite `JUnitWrapperSuite`class. This eventually sends a `SuiteStarted` event to the reporter with this class name.

For the tests themselves, we parse the suite name [from their description using a regex](https://github.com/scalatest/scalatest/blob/3.0.x/scalatest/src/main/scala/org/scalatest/junit/MyRunListener.scala#L103). This ends up sending events like `TestStarting` with the `suiteId` of the actual JUnit class name (in my example that would be `ExampleSuite`). 

The XML reporter seems to ignore tests from suites it hasn't seen before (based on the `SuiteStarted` event).

A potential fix is to override `suiteId` to `junitClassName` on [JUnitWrapperSuite](https://github.com/scalatest/scalatest/blob/3.0.x/scalatest/src/main/scala/org/scalatest/junit/JUnitWrapperSuite.scala#L49).

I'm happy to submit a PR, but I'm not sure how to bubble this up to newer versions. On 3.2.2, the runner seems to be completely broken for JUnit suites, see #1971.

## Versions
```
<dependencies>
    <dependency>
      <groupId>org.scalactic</groupId>
      <artifactId>scalactic_2.13</artifactId>
      <version>3.0.8</version>
    </dependency>
    <dependency>
      <groupId>org.scalatest</groupId>
      <artifactId>scalatest_2.13</artifactId>
      <version>3.0.8</version>
    </dependency>
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>4.13.2</version>
    </dependency>
  </dependencies>
  ```

**Comment by iuliand-db** [comment]
@cheeseng do you think this is something that could be fixed, or is there a good workaround?

**Comment by cheeseng** [comment]
@iuliand-db @gabrielrussoc I am not sure what's wrong yet at the moment, but I think it will be good to try to reproduce it under the just released 3.2.5, I think in order to call org.scalatest.tools.Runner.main directly to run junit suites you need to have the scalatest+junit jar on the classpath.  In issue 1971 I think you may try to change the scalatest+junit version to 3.2.2.0.

**Comment by cheeseng** [comment]
@gabrielrussoc Yes I think using junitClassName as suiteId is a good idea, I just submitted the following PR: 

https://github.com/scalatest/scalatestplus-junit/pull/24

Let's see if @bvenners thinks it is a good idea, and hopefully it can be released as scalatest+junit 3.2.6.0.

**Comment by gabrielrussoc** [comment]
awesome, thanks a lot

would it be possible to port this to 3.0.x?

**Comment by bvenners** [comment]
@gabrielrussoc We'd rather not go that far back. Is there some tool or library you are using that is stuck on ScalaTest 3.0.x?

**Comment by gabrielrussoc** [comment]
@bvenners we have scalatest 3.0.x in our codebase and bumping it to 3.2 takes some effort since there was a lot of packages moved around.

In theory we can upgrade, but since this is a fairly trivial change, I was hoping we could just port it to 3.0 as well.

**Comment by bvenners** [comment]
@gabrielrussoc We released ScalaFix tools to automatically fix the deprecation warnings, described on the readme on these pages:

https://github.com/scalatest/autofix/tree/master/3.0.x

And:

https://github.com/scalatest/autofix/tree/master/3.1.x

It takes us a non-trivial amount of time to do a release these days, because there are so many versions and variants of Scala that ScalaTest supports, so I'd rather convince you to upgrade.

**Comment by gabrielrussoc** [comment]
@bvenners Is there a way to run those scripts standalone? I see they are provided as plugins for Maven, mill and sbt but we use Bazel :(
